sopel/tools/__init__.py

```diff
@@ -769,3 +769,41 @@ def get_raising_file_and_line(tb=None):
     filename, lineno, _context, _line = traceback.extract_tb(tb)[-1]
 
     return filename, lineno
+
+
+def chain_loaders(*lazy_loaders):
+    """Chain lazy loaders into one.
+
+    :return: a lazy loader that combines all of the given ones
+    :rtype: :term:`function`
+
+    This function takes any number of lazy loaders as arguments and merge them
+    together into one. This loader can be used in lazy rule decorators, such as
+    the :func:`sopel.plugin.url_lazy` decorator::
+
+        import plugin
+
+        def _lazy_spam(settings):
+            # figure out spam rules
+
+        def _lazy_eggs(settings):
+            # figure out eggs rules
+
+        @plugin.url_lazy(
+            plugin.chain_loaders(_lazy_spam, _lazy_eggs))
+        def spam_and_eggs(bot, trigger):
+            # whatever happens here
```

Comment:
```suggestion
    This function takes any number of lazy loaders as arguments and merges them
    together into one. It's primarily a helper for lazy rule decorators such as
    :func:`sopel.plugin.url_lazy`.
```

We don't need that old example any more now that `@plugin.url_lazy` uses this function itself, automatically.